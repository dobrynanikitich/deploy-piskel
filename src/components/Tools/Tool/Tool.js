import React from 'react';
import classnames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import classes from './Tool.module.scss';

const Tool = ({ active, icon, setActiveTool, id, tooltip }) => {
    return (
        <div
            className={classnames(classes.Tool, { [`${classes.ToolActive}`]: active })}
            id={id}
            onClick={setActiveTool}
            data-tip={tooltip}
        >
            <FontAwesomeIcon icon={icon} size="2x"/>
        </div>
    );
}

export default Tool;
