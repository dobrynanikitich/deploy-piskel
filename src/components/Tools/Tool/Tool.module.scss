.Tool {
    width: 60px;
    height: 60px;
    background: darkslategray;
    box-sizing: border-box;
    display: flex;
    justify-content: center;
    align-items: center;
    color: white;
}

.ToolActive {
    border: 3px solid goldenrod;
}