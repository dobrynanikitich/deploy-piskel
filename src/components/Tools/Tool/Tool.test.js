import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import sinon from 'sinon';

import Tool from './Tool';

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const wrapper = shallow(
        <Tool
        id={0}
        key={0}
        active={true}
        icon={'pen'}
        setActiveTool={() => {}}
        tooltip={'pen'}
         />
    );
    expect(wrapper).toMatchSnapshot();
});

it('renders correctly dom', () => {
    const wrapper = render(
        <Tool
            id={0}
            key={0}
            active={true}
            icon={'pen'}
            setActiveTool={() => {}}
            tooltip={'pen'}
         />
    );
    expect(wrapper).toMatchSnapshot();
});

it('back button doesnt calls', () => {
    const spy = sinon.spy();
    const wrapper = mount(
        <Tool
            id={0}
            key={0}
            active={true}
            icon={'pen'}
            setActiveTool={spy}
            tooltip={'pen'}
         />
    );
  
    wrapper.simulate('click');
  
    expect(spy.calledOnce).toBe(true)
});