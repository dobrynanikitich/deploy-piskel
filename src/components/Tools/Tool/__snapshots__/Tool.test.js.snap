// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Tool ToolActive"
  data-tip="pen"
  id={0}
  onClick={[Function]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="pen"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="2x"
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;

exports[`renders correctly dom 1`] = `
<div
  class="Tool ToolActive"
  data-tip="pen"
  id="0"
/>
`;
